components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/JDBCPermissionBasedInternalScopeValidator.java

```diff
@@ -406,31 +402,15 @@ private String[] getAllowedResourcesOfUser(AuthenticatedUser authenticatedUser,
         return (String[]) ArrayUtils.add(allowedUIResourcesForUser, EVERYONE_PERMISSION);
     }
 
-    private String[] retrieveUserOrganizationPermission(AuthenticatedUser authenticatedUser, String organizationId) {
+    private String[] retrieveUserOrganizationPermission(AuthenticatedUser authenticatedUser, String organizationId)
+            throws UserIdNotFoundException {
 
         //Add permission based on user's organization roles.
         String[] allowedUIResourcesForUser = null;
         if (StringUtils.isNotBlank(organizationId)) {
             try {
-                List<String> organizationPermissions = new ArrayList<>();
-                String authenticatedUserId = null;
-                try {
-                    authenticatedUserId = authenticatedUser.getUserId();
-                } catch (UserIdNotFoundException e) {
-                    // Resolve authenticated resident user.
-                    Optional<User> resolvedUser =
-                            OAuth2ServiceComponentHolder.getOrganizationUserResidentResolverService()
-                                    .resolveUserFromResidentOrganization(
-                                            authenticatedUser.getUserName(), null, organizationId);
-                    if (resolvedUser.isPresent()) {
-                        authenticatedUserId = resolvedUser.get().getUserID();
-                        authenticatedUser.setUserId(authenticatedUserId);
-                    }
-                }
```

Comment:
This code block should be obsolete now. As user resolve from resident organization logics are not required as user's resident organization is managed within authenticated user object, which aid to resolve the userID when ever necessary without throwing UserIdNotFoundException.